jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedGarbageCollectorMXBeanImpl.java

```diff
@@ -81,12 +83,15 @@ static GcInfo buildGcInfo(long index, long startTime, long endTime,
 							long[] initialSize, long[] preUsed, long[] preCommitted, long[] preMax,
 							long[] postUsed, long[] postCommitted, long[] postMax) {
 		/* retrieve the names of MemoryPools*/
-		List<MemoryPoolMXBean> memoryPoolList = ManagementFactory.getMemoryPoolMXBeans();
-		String[] poolNames = new String[memoryPoolList.size()];
-		int idx = 0;
-		for (MemoryPoolMXBean bean : memoryPoolList) {
-			poolNames[idx++] = bean.getName();
+		if (null == poolNames) {
+			List<MemoryPoolMXBean> memoryPoolList = ExtendedMemoryMXBeanImpl.getInstance().getMemoryPoolMXBeans(false);
+			poolNames = new String[memoryPoolList.size()];
+			int idx = 0;
+			for (MemoryPoolMXBean bean : memoryPoolList) {
+				poolNames[idx++] = bean.getName();
+			}
```

Comment:
Using a stream this could be simpler:
```java
		poolNames = ExtendedMemoryMXBeanImpl.getInstance().getMemoryPoolMXBeans(false)
				.stream()
				.map(MemoryPoolMXBean::getName)
				.toArray(String[]::new);
```